src/com/TTT/TurnUI.java

```diff
@@ -1,14 +1,47 @@
 package com.TTT;
 
 import java.io.InputStream;
+import java.io.PrintStream;
 import java.util.Scanner;
 
 public class TurnUI {
 
-    public void takeTurn(Board board, BoardPrinter printer, Game game, InputStream in) {
+    private PrintStream out;
+    private BoardPrinter printer;
+    private Scanner scanner;
+    private String inputPrompt ="Please enter a space number: ";
+
+    public TurnUI(BoardPrinter printer, PrintStream out, InputStream in) {
+        this.printer = printer;
+        this.out = out;
+        scanner = new Scanner(in);
+        scanner.useDelimiter("\n");
+    }
+
+    public void takeTurn(Board board, Game game) {
+        int selectedSpace = getPlayerInput(board, inputPrompt);
+
+        while(selectedSpace < 0 || selectedSpace > (board.size() - 1) || board.isOccupied(selectedSpace))
+        {
+            selectedSpace = getPlayerInput(board, "Invalid input. " + inputPrompt);
+        }
+
+        game.doTurn(selectedSpace);
+    }
+
+    private Integer getPlayerInput(Board board, String text) {
         printer.printBoard(board);
-        Scanner sc = new Scanner(in);
-        String selection = sc.nextLine();
-        game.doTurn(Integer.parseInt(selection));
+        out.print(text);
+        String selection = scanner.nextLine();
+        return parseSelection(selection);
+    }
+
+    private int parseSelection(String selection) {
+        try{
+            return Integer.parseInt(selection) - 1;
+        }
+        catch(Exception e){
```

Comment:
I'd avoid catching a type so high up the inheritance tree. Can you handle the specific exception that you expect here?